Index: src/Fonctions/fonctions.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\r\nnamespace App\\Fonctions;\r\n    function Redirect_Self_URL():void{\r\n        unset($_REQUEST);\r\n        header(\"Location: \".$_SERVER['PHP_SELF']);\r\n        exit;\r\n    }\r\n\r\nfunction GenereMDP($nbChar) :string{\r\n\r\n    return \"secret\";\r\n}\r\n\r\nfunction CalculComplexiteMdp($mdp) :int{\r\n\r\n\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Fonctions/fonctions.php b/src/Fonctions/fonctions.php
--- a/src/Fonctions/fonctions.php	(revision 2ccf27da20903b9a1aeb0adb29db27f87992f44b)
+++ b/src/Fonctions/fonctions.php	(date 1731506007394)
@@ -1,5 +1,6 @@
 <?php
 namespace App\Fonctions;
+
     function Redirect_Self_URL():void{
         unset($_REQUEST);
         header("Location: ".$_SERVER['PHP_SELF']);
@@ -11,7 +12,15 @@
     return "secret";
 }
 
-function CalculComplexiteMdp($mdp) :int{
+    function CalculComplexiteMdp($mdp) :int {
+        $L = strlen($mdp); // Longueur du mot de passe
 
+        $B = 0;
+        if (preg_match('/[a-z]/', $mdp)) $B += 26; // Lettres minuscules
+        if (preg_match('/[A-Z]/', $mdp)) $B += 26; // Lettres majuscules
+        if (preg_match('/[0-9]/', $mdp)) $B += 10; // Chiffres
+        if (preg_match('/[^a-zA-Z0-9]/', $mdp)) $B += 10; // Caractères spéciaux
 
-}
\ No newline at end of file
+        $E = $L * log($B, 2); // Entropie en bits
+        return ($E);
+    }
\ No newline at end of file
